src/include/duckdb/execution/operator/join/physical_cross_product.hpp

```diff
@@ -9,18 +9,23 @@
 #pragma once
 
 #include "duckdb/common/types/chunk_collection.hpp"
-#include "duckdb/execution/physical_operator.hpp"
+#include "duckdb/execution/physical_sink.hpp"
 
 namespace duckdb {
 //! PhysicalCrossProduct represents a cross product between two tables
-class PhysicalCrossProduct : public PhysicalOperator {
+class PhysicalCrossProduct : public PhysicalSink {
 public:
 	PhysicalCrossProduct(vector<LogicalType> types, unique_ptr<PhysicalOperator> left,
 	                     unique_ptr<PhysicalOperator> right);
 
+	std::mutex rhs_lock;
```

Comment:
The lock should be moved tot he GlobalOperatorState; as a rule the operator itself should only contain the constants necessary to describe what needs to be done (e.g. expressions, join conditions, etc). Any tools necessary for performing the operation (intermediates, locks, etc) should be in the operator states.

Reply:
Good point. Done!